Tutorial for wrapping C++ operations for user in C++ and Python

arthurp opened the pull request.

arthurp committed:
Clean up existing algorithms.

I needed to solidify the conventions in my mind and I needed
simple examples.

arthurp committed:
Add document describing the graph API design.

gurbinder533 commented on docs/GraphOperationAPIs.md:
````diff
@@ -0,0 +1,739 @@
+### Complete Example: SSSP
+
+```c++
+#ifndef KATANA_LIBGALOIS_KATANA_ANALYTICS_SSSP_SSSP_H_
````
Do we need the complete example here? Can we just have a pointer to the file?

arthurp replied:
Yea. You are probably right. I'll change it.

gurbinder533 reviewed (approved):
LGTM.

arthurp committed:
Clean up doc build target and check_docs.sh.

It would produce lots of spurious warnings when run in a dirty
build directory and the check script would output any warnings it
detected twice.

arthurp commented:
@bowu, could I get a review on this. I had a change a few things since Gurbinder reviewed and I would like confirmation that this mostly makes sense based on your experiences. Obviously this document still needs work. But I want to get it in the repo anyway.

bowu reviewed (approved):
LGTM.
